src/components/_common/Input/Input.tsx

```diff
@@ -0,0 +1,212 @@
+import styled from "@emotion/styled";
+import { useState, KeyboardEvent } from "react";
+import InputUpload from "./InputUpload";
+
+interface InputProps {
+  width?: string;
+  height?: string;
+  placeholder: string;
+  isTextarea?: boolean;
+  hasTags?: boolean;
+  tags?: string[];
+  hasImage?: boolean;
+  image?: string;
+}
+
+interface IInputStyle{
+  width: string;
+  height: string;
+}
+
+export const Input = ({
+  width = '100%',
+  height = 'auto',
+  placeholder,
+  isTextarea = false,
+  hasTags = false,
+  tags: initialTags = [],
+  hasImage = false,
+  image: initialImage,
+}: InputProps) => {
+  const [inputValue, setInputValue] = useState("");
+  const [tags, setTags] = useState(initialTags);
+  const [image, setImage] = useState<string | null>(initialImage || null);
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    if (event.key === 'Enter' && inputValue.trim() && hasTags) {
+      setTags([...tags, inputValue.trim()]);
+      setInputValue('');
+    }
+  };
+
+  // TODO: 추후 데이터 관련하여 다시 확인..
+  const handleImageChange = (uploadedFile: File) => {
+    setImage(URL.createObjectURL(uploadedFile));
+  };
+
+  const handleTagRemove = (indexToRemove: number) => {
+    setTags(tags.filter((_, index) => index !== indexToRemove));
+  };
+
+  const handleImageRemove = () => {
+    setImage(null);
+  };
+
+  const handleImageZoom = () => {
+    // TODO: 이미지 확대
+  };
+
+  return (
+    <StInputContainer width={width} height={height}>
+      {isTextarea ? (
+        <StTextArea
+          value={inputValue}
+          onChange={(e) => setInputValue(e.target.value)}
+          onKeyDown={handleKeyDown}
+          placeholder={placeholder}
+        />
+      ) : (
+        <StInputText
+          value={inputValue}
+          onChange={(e) => setInputValue(e.target.value)}
+          onKeyDown={handleKeyDown}
+          placeholder={placeholder}
+        />
+      )}
+
+      {/* TODO: 태그 컴포넌트 병합시 진행 */}
+      {hasTags && (
+        <StTagsContainer>
+          {tags.map((tag, index) => (
+            <StTag key={index}>
+              {tag}
+              <StTagRemoveButton onClick={() => handleTagRemove(index)}>x</StTagRemoveButton>
+            </StTag>
+          ))}
+        </StTagsContainer>
+      )}
+
+      {/* TODO: Image 클릭 시 확장 어떻게 진행할지 */}
+      {hasImage && (
+        !image ? (
+          <StImageContainer>
+            <StyledUpload>
+              <InputUpload onChange={handleImageChange}>+</InputUpload>
+            </StyledUpload>
+          </StImageContainer>
+        ) : (
+          <StImageContainer>
+            <StImage style={{ backgroundImage: `url(${image})` }} onClick={handleImageZoom}>
+              <StImageDeleteButton onClick={handleImageRemove}>x</StImageDeleteButton>
+            </StImage>
+          </StImageContainer>
+        )
+      )}
```

Comment:
```suggestion
      {hasImage && (
          <StImageContainer>
            { image ? (
                <StImage 
                  style={{ backgroundImage: `url(${image})` }}
                  onClick={handleImageZoom}>
                  <StImageDeleteButton 
                     onClick={handleImageRemove}>
                     x
                   </StImageDeleteButton>
                </StImage>
                )  :  (
                <StyledUpload>
                     <InputUpload 
                          onChange={handleImageChange}>
                          +
                      </InputUpload>
                </StyledUpload>
                )}
          </StImageContainer>
        )} 
```
`StImageContainer`까지 리렌더링할 필요는 없어보여서 분리해봤습니다!

그리고 prettier 적용이 안되시는것 같아요!

`singleAttributePerLine`으로 속성들이 한줄에 하나씩만 들어가야합니다~!